tests/personal_data/portfolios/types/test_future_portfolio.py

```diff
@@ -348,6 +348,121 @@ async def test_update_portfolio_data_from_order_with_cancelled_and_filled_orders
     assert portfolio_manager.portfolio.get_currency_portfolio("USDT").total == decimal.Decimal('999.616')
 
 
+async def test_update_portfolio_data_from_orders_with_max_long_size_linear_contract(
+        future_trader_simulator_with_default_linear):
+    # TODO same test with fees in orders
+    config, exchange_manager_inst, trader_inst, default_contract = future_trader_simulator_with_default_linear
+    portfolio_manager = exchange_manager_inst.exchange_personal_data.portfolio_manager
+    default_contract.set_current_leverage(decimal.Decimal(2))
+
+    # Test buy order
+    market_buy = BuyMarketOrder(trader_inst)
+    buy_order_price = decimal.Decimal(40)
+    market_buy.update(order_type=enums.TraderOrderType.BUY_MARKET,
+                      symbol=DEFAULT_FUTURE_SYMBOL,
+                      current_price=buy_order_price,
+                      quantity=decimal.Decimal(50),
+                      price=buy_order_price)
+
+    # Test sell order
+    market_sell = SellMarketOrder(trader_inst)
+    sell_order_price = decimal.Decimal(80)
+    market_sell.update(order_type=enums.TraderOrderType.SELL_MARKET,
+                       symbol=DEFAULT_FUTURE_SYMBOL,
+                       current_price=sell_order_price,
+                       quantity=decimal.Decimal(50),
+                       price=sell_order_price)
+
+    portfolio_manager.portfolio.update_portfolio_available(market_buy, True)
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").available == constants.ZERO
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").total == decimal.Decimal('1000')
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").order_margin == decimal.Decimal('1000')
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").position_margin == constants.ZERO
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").unrealized_pnl == constants.ZERO
+
+    # Open long position
+    await fill_market_order(market_buy)
+    await _update_position_mark_price(exchange_manager_inst, sell_order_price, DEFAULT_FUTURE_SYMBOL)
+    assert portfolio_manager.portfolio.get_currency_portfolio("USDT").available == constants.ZERO
```

Comment:
I don't think this is possible (`self.available = self.wallet_balance - self.position_margin - self.order_margin`)